projects/dev/src/app/group/group-manage/pending-request/pending-request.component.ts

```diff
@@ -22,87 +45,141 @@ export class PendingRequestComponent implements OnInit, OnChanges {
     { field: "at", order: -1 },
     { field: "member_id", order: 1 },
   ];
-  prevSortMeta = "-at member_id";
-
-  @Output() onAccept = new EventEmitter<any>();
-  @Output() onReject = new EventEmitter<any>();
-
-  groupSwitch = [
-    {
-      label: "This group only",
-    },
-    {
-      label: "All subgroups",
-    },
-  ];
+  prevSortMeta: string[] = GROUP_REQUESTS_API.sort;
+
+  requestAction: AcceptReject = AcceptReject.None;
+  selection = [];
```

Comment:
This variable (and other above) still has no type, yes TS knows it is an array but of "any" while only some values are authorized and are handled in the html.

Reply:
fixed.